ipr/summary.py

```
@@ -278,11 +281,20 @@ def get_preferred_gene_name(graphkb_conn: GraphKBConnection, record_id: str) ->
 
 def display_variants(gene_name: str, variants: List[IprVariant]):
     def display_variant(variant: IprVariant):
-        if 'proteinChange' in variant:
+        if 'proteinChange' in variant and variant['proteinChange']:
             return f'{variant["gene"]}:{variant["proteinChange"]}'
-        if 'gene1' in variant:
-            return f'({variant["gene1"]},{variant["gene2"]}):fusion(e.{variant["exon1"]},e.{variant["exon2"]})'
-        return f'{variant["kbCategory"]} of {variant["gene"]}'
+        if 'gene1' in variant and 'gene2' in variant:
+            try:
+                return f'({variant["gene1"]},{variant["gene2"]}):fusion(e.{variant["exon1"]},e.{variant["exon2"]})'
+            except KeyError as err:
+                logger.error(f'Malformed fusion {err} in {variant}')
+                return f'({variant["gene1"]},{variant["gene2"]}):fusion(e.?,e.?)'
+        if 'kbCategory' in variant and variant['kbCategory']:
+            return f'{variant["kbCategory"]} of {variant["gene"]}'
+        if 'variant' in variant:
+            logger.warning(f'Unexpected format variant {variant["variant"]}: {variant}')
+            return variant['variant']
+        logger.error("Could not display {variant}")
```

this should be a breaking error rather than a warning, we don't want to continue upload if the summary generation fails

---

raised a ValueError instead

---

I believe this same error was also referred to as GERO-53